python/feathub/feature_views/derived_feature_view.py

```diff
@@ -98,12 +104,24 @@ def build(
                 feature_descriptors=[self.source], force_update=force_update
             )[0]
 
+        default_field_name_index = 0
+        while f"_{default_field_name_index}" in [
+            x.name for x in source.get_output_features()
+        ]:
+            default_field_name_index += 1
+
         features = []
         for feature in self.features:
             if isinstance(feature, str):
                 feature = self._get_feature_from_feature_str(
-                    feature, registry, source, force_update
+                    feature,
+                    registry,
+                    source,
+                    force_update,
+                    f"_{default_field_name_index}",
                 )
+                if feature.name == f"_{default_field_name_index}":
+                    default_field_name_index += 1
```

Comment:
It is possible that the this feature name collide with source.get_output_features().

Maybe we should put the logic above (which uses a while loop) into a method and call the method here?